mesh/mesh.cpp

```diff
@@ -7283,28 +7283,14 @@ void Mesh::UniformRefinement(int ref_algo)
    {
       NURBSUniformRefinement();
    }
-   else if (ref_algo == 0 && Dim == 3 && meshgen == 1)
-   {
-      UniformRefinement3D();
-   }
-   else if (meshgen == 1 || ncmesh)
+   else if (ncmesh)
```

Comment:
The modifications here ignore the case `ref_algo == 1` which should apply only for conforming tet-only meshes -- in this case we want to do bisection-based tet refinement of all elements using `LocalRefinement`.

Reply:
Thanks, I'll look into it.

Reply:
I put back the branch `ref_algo == 1 && meshgen == 1`. The branch is now at the same level (in the method) as the other cases, so it doesn't get lost in the future (that's why the refinement list code is duplicated).